consensus/src/quorum_store/utils.rs

```diff
@@ -427,25 +468,29 @@ impl ProofQueue {
         info!(
             // before non full check
             byte_size = cur_bytes,
-            block_size = cur_txns,
+            block_total_txns = cur_all_txns,
+            block_unique_txns = cur_unique_txns,
             batch_count = ret.len(),
             full = full,
             return_non_full = return_non_full,
             "Pull payloads from QuorumStore: internal"
         );
 
         if full || return_non_full {
-            counters::BLOCK_SIZE_WHEN_PULL.observe(cur_txns as f64);
+            counters::BLOCK_SIZE_WHEN_PULL.observe(cur_unique_txns as f64);
+            counters::TOTAL_BLOCK_SIZE_WHEN_PULL.observe(cur_all_txns as f64);
+            counters::KNOWN_DUPLICATE_TXNS_WHEN_PULL
+                .observe((cur_all_txns - cur_unique_txns) as f64);
```

Comment:
@igor-aptos I've added the counters for number of unique transactions and the number of all transactions here.

Reply:
that's on pull (i.e. on block proposer). I am suggesting equivalent counter that all nodes report after they dedup transactions during execution (if that doesn't exist already)

Reply:
This might be what you are looking for?
https://github.com/aptos-labs/aptos-core/blob/satya/pull_proofs/consensus/src/block_preparer.rs#L66